traffic_ops/app/db/create_tables.sql

```diff
@@ -143,6 +143,42 @@ LANGUAGE plpgsql;
 
```

Comment:
Wait, those `DNS-SEC:*` Permissions are coming from a migration that hasn't been squashed. The only way they could be missing is if you didn't run migrations, had a pre-existing Role with the name `admin` that has a priv_level under 30, or modified the database after you were done using `github.com/apache/trafficcontrol/traffic_ops/app/db/admin`.

Reply:
`PARAMETER-SECURE:READ` is specifically being added in a migration to any users with the same "PrivLevel" as "admin" (has by default). It's not appropriate to include data in the schema creation, and it's not appropriate to modify a system that may be using the Permissions security system instead of PrivLevel by granting users a Permission they didn't have before seeding; seeding should only add data not update it.

The only place it makes sense to me to be granting that Permission is in a migration. "admin" is intentionally only given the `ALL` Permission in `seeds.sql` because it always has all Permissions, regardless of any introduced or removed in the future. System administrators are meant to have to manually assign new Permissions to their users, so if they are doing a new install, it's correct for "admin" to not have the Permission because it only has "ALL". If they're updating, then they would have had to run the now-squashed migration, so any user that needs it will have it, including "admin".

tl;dr - I don't think it's a bug that `admin` is missing those Permissions.

Reply:
On a related note, the `/deliveryservices/{{ID}}/safe` endpoint requires an "Unauthenticated" privilige level. Removal of the ability by default to perform a non-read-only action from a Role named "read-only" which is meant to only perform read-only actions is intentional, and was reflected in `seeds.sql` prior to this PR, despite that users from an existing installation with the Role by the same name has the ability to do so by virtue of their "Privilege Level".

Reply:
> Wait, those `DNS-SEC:*` Permissions are coming from a migration that hasn't been squashed. The only way they could be missing is if you didn't run migrations, had a pre-existing Role with the name `admin` that has a priv_level under 30, or modified the database after you were done using `github.com/apache/trafficcontrol/traffic_ops/app/db/admin`.

While you are right that those perms are being added in a migration, the migrations are definitely being run, because I am using the `db/admin` tool to do it (see https://github.com/apache/trafficcontrol/pull/6802#issuecomment-1133448159 for the script I am using).

From my testing, the failure for those perms to exist once those migrations are run does seem to be a result of the squash in #6802.

Reply:
It appears that the issue is that resetting the database runs the table creation script, followed by all migrations. Data seeding is never done, and the migration that adds the Permissions in question has already run so a following `upgrade` command won't run it.

However, the state of the database at that point is perfectly valid. The `admin` Role exists and has the special `ALL` Permission (which actually does nothing, and anyway an `admin`-Role user's Permissions are never checked). Any existing users with equivalent Permissions to `admin` will have the Permissions added to keep their ability to access the API up-to-date with `admin`.

If anything, I'd say the migration should specifically *not* add any Permissions to the special `admin` Role; it's meant to be immutable, and in fact the API itself will not allow adding Permissions to that Role. But existing migrations shouldn't be edited. Best I can think of doing is adding a statement to the patches script that strips `admin` of all Permissions besides `ALL` (and perhaps removes `ALL` from any and all other Roles that may have it). That seems somewhat unnecessary to me, but if you think it's a good idea then I'll do it.